src/common/date-helpers.ts

```diff
@@ -0,0 +1,48 @@
+export function getDayNames (firstDay: number = 0, nameLength: string = 'short', locale: string = 'en-US'): Array<string> {
+    // Days start from Sunday (Sunday = 0, Monday = 1, etc.)
+    const listOfDayNames: Array<string> = [];
+
+    // For each of the seven days of the week
+    for (let i = 0; i < 7; i++) {
+        let dayName = new Date(1970, 5, i + firstDay).toLocaleString(locale, { weekday: nameLength });
+        listOfDayNames.push(dayName);
+    }
+
+    return listOfDayNames;
+}
```

Comment:
I believe creating array just for the sake of index count is making the code more obscure and less efficient with no actual gain. If we were using lodash, i would perhaps use `lodash.times().map()`. But the current approach is totally fine.